aii-core/src/main/perl/Shellfe.pm

```diff
@@ -323,6 +339,38 @@ sub _initialize
     if ($self->option(INCLUDE)) {
         unshift(@INC, split(/:+/, $self->option(INCLUDE)));
     }
+
+    # Initialization and validation of some options whose default values are based on other options
+    if ( $self->option(NBPDIR_PXELINUX) ) {
+        $self->{CONFIG}->set(NBPDIR_PXELINUX, undef) if $self->option(NBPDIR_PXELINUX) eq NBPDIR_VARIANT_DISABLED;
+    }
+    if ( $self->option(NBPDIR_GRUB2) ) {
+        $self->{CONFIG}->set(NBPDIR_GRUB2, undef) if $self->option(NBPDIR_GRUB2) eq NBPDIR_VARIANT_DISABLED;
```

Comment:
use a variable to defined the value in the if/else blocks, and set it once afterwards. it is really not easy to review duplicate code.